app/controllers/admin/content/events.js

```diff
@@ -0,0 +1,128 @@
+import Controller from '@ember/controller';
+import { computed } from '@ember/object';
+import { later } from '@ember/runloop';
+
+export default Controller.extend({
+
+  subTopics: computed('model.topic', function() {
+    later(this, () => {
+      try {
+        this.set('subTopic', null);
+      } catch (ignored) { /* To suppress error thrown in-case this gets executed after component gets destroy */ }
+    }, 50);
+    if (!this.get('model.topic')) {
+      return [];
+    }
+    return this.get('model.topic.subTopics');
+  }),
+
+  actions: {
```

Comment:
Too much repetition of very similar stuff. 

`openNewEventTypeModal`, `openNewEventTopicModal`, `openNewEventSubTopicModal` can become one action
`deleteEventType`, `deleteEventTopic`, `deleteEventSubTopic` can become one action
`addEventType`, `addEventTopic`, `addEventSubTopic` can become one action.